clients/settings/permissions/permissions.go

```diff
@@ -0,0 +1,163 @@
+	return &Client{client: client}
+}
+
+func (c *Client) GetAllAccessors(ctx context.Context, objectID string) (api.Response, error) {
+	return c.get(ctx, objectID, "", "")
+}
+
+func (c *Client) GetAllUsersAccessor(ctx context.Context, objectID string) (api.Response, error) {
+	return c.get(ctx, objectID, allUsersAccessorType, "")
+}
+
+func (c *Client) GetAccessor(ctx context.Context, objectID string, accessorType string, accessorID string) (api.Response, error) {
+	if accessorType == "" {
+		return api.Response{}, ErrorPermissions{Wrapped: ErrorMissingAccessorType, Operation: GET}
+	}
+
+	if accessorID == "" {
+		return api.Response{}, ErrorPermissions{Wrapped: ErrorMissingAccessorID, Operation: GET}
+	}
+
+	return c.get(ctx, objectID, accessorType, accessorID)
+}
+
+func (c *Client) get(ctx context.Context, objectID string, accessorType string, accessorID string) (api.Response, error) {
+	if objectID == "" {
+		return api.Response{}, ErrorPermissions{Wrapped: ErrorMissingObjectID, Operation: GET}
+	}
+
+	path, err := url.JoinPath(endpointConfigPath, objectID, permissionResourcePath, accessorType, accessorID)
+	if err != nil {
+		return api.Response{}, ErrorPermissions{Wrapped: err, Operation: GET}
+	}
+
+	resp, err := c.client.GET(ctx, path, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+
+	if err != nil {
+		return api.Response{}, ErrorPermissions{Wrapped: err, Operation: GET}
+	}
+
+	return api.NewResponseFromHTTPResponse(resp)
+}
+
+func (c *Client) Create(ctx context.Context, objectID string, body []byte) (api.Response, error) {
+	if objectID == "" {
+		return api.Response{}, ErrorPermissions{Wrapped: ErrorMissingObjectID, Operation: POST}
+	}
+
+	path, err := url.JoinPath(endpointConfigPath, objectID, permissionResourcePath)
+	if err != nil {
+		return api.Response{}, ErrorPermissions{Wrapped: err, Operation: POST}
+	}
+
+	resp, err := c.client.POST(ctx, path, bytes.NewReader(body), rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+	if err != nil {
+		return api.Response{}, ErrorPermissions{Wrapped: err, Operation: POST}
+	}
+
+	return api.NewResponseFromHTTPResponse(resp)
+}
+
+func (c *Client) UpdateAllUsersAccessor(ctx context.Context, objectID string, body []byte) (api.Response, error) {
+	return c.update(ctx, objectID, allUsersAccessorType, "", body)
+}
+
+func (c *Client) UpdateAccessor(ctx context.Context, objectID string, accessorType string, accessorID string, body []byte) (api.Response, error) {
+	if accessorType == "" {
+		return api.Response{}, ErrorPermissions{Wrapped: ErrorMissingAccessorType, Operation: PUT}
+	}
+
+	if accessorID == "" {
+		return api.Response{}, ErrorPermissions{Wrapped: ErrorMissingAccessorID, Operation: PUT}
+	}
+
+	return c.update(ctx, objectID, accessorType, accessorID, body)
+}
+
+func (c *Client) update(ctx context.Context, objectID string, accessorType string, accessorID string, body []byte) (api.Response, error) {
+	if objectID == "" {
```

Comment:
These are real Java-style asserts and Go usually does not do that. I would remove them tbh.

@tomazpu, could you maybe bring in a 2nd opinion here (or 3rd :D)